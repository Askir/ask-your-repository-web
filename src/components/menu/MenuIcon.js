// @flow
import React from 'react';
import './Menu.scss';

type Props = {
  open: boolean,
};

function MenuIcon({ open, ...props }: Props) {
  const menuClass = open ? 'burger-menu open' : 'burger-menu';
  return (
    <div className={menuClass} {...props}>
      <div className="bar1" key="b1" />
      <div className="bar2" key="b2" />
      <div className="bar3" key="b3" />
    </div>
  );
}

export default MenuIcon;
