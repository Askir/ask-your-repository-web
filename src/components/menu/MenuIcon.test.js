// @flow
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MenuIcon from './MenuIcon';

describe('<MenuIcon />', () => {
  let wrapper;

  beforeEach(() => {
    const open = false;
    wrapper = shallow(<MenuIcon open={open} />);
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('Props open is true', () => {
    beforeEach(() => {
      const open = true;
      wrapper = shallow(<MenuIcon open={open} />);
    });

    it('uses styleclass burger-menu open', () => {
      expect(wrapper.find('#burger_icon').hasClass('burger-menu open')).toEqual(true);
    });
  });

  describe('Props open is false', () => {
    beforeEach(() => {
      const open = false;
      wrapper = shallow(<MenuIcon open={open} />);
    });

    it('uses styleclass burger-menu', () => {
      expect(wrapper.find('#burger_icon').hasClass('burger-menu')).toEqual(true);
    });
  });
});
