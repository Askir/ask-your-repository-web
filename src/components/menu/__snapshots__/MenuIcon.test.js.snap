// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuIcon /> renders correctly 1`] = `
<div
  className="burger-menu"
  id="burger_icon"
>
  <div
    className="bar1"
    key="b1"
  />
  <div
    className="bar2"
    key="b2"
  />
  <div
    className="bar3"
    key="b3"
  />
</div>
`;
